src/python/pants/python/python_setup.py

```diff
@@ -204,15 +209,18 @@ def artifact_cache_dir(self):
     def scratch_dir(self):
         return os.path.join(self.get_options().pants_workdir, *self.options_scope.split("."))
 
-    def compatibility_or_constraints(self, compatibility):
```

Comment:
Note that this is a private API, so this change is safe to do without deprecation.